kobo/apps/stripe/tests/test_organization_usage.py

```diff
@@ -509,3 +501,36 @@ def test_get_suscription_limit_unlimited(self, usage_type):
         generate_plan_subscription(self.organization, metadata=product_metadata)
         limit = get_organization_plan_limit(self.organization, usage_type)
         assert limit == float('inf')
+
+
+@override_settings(STRIPE_ENABLED=True)
+class OrganizationsModelIntegrationTestCase(BaseTestCase):
+    fixtures = ['test_data']
+
+    def setUp(self):
+        self.someuser = User.objects.get(username='someuser')
+        self.organization = self.someuser.organization
+
+    def test_is_mmo_subscription_logic(self):
+        self.organization.mmo_override = True
+        self.organization.save()
+        assert self.organization.is_mmo is True
+
+        self.organization.mmo_override = False
+        self.organization.save()
+        assert self.organization.is_mmo is False
+
+        product_metadata = {
+            'mmo_enabled': 'false',
+        }
+        subscription = generate_plan_subscription(
+            self.organization, metadata=product_metadata
+        )
+        assert self.organization.is_mmo is False
+        subscription.status = 'canceled'
+        subscription.ended_at = timezone.now()
+        subscription.save()
+
+        product_metadata['mmo_enabled'] = 'true'
+        generate_plan_subscription(self.organization, metadata=product_metadata)
+        assert self.organization.is_mmo is True
```

Comment:
Is there any case where - in the same request - `is_mmo` could change from False to True? 
I'm wondering if `cache_for_request` could be a problem and we would need to void the cache for request when new subscription is created?

Reply:
Currently, as far as I know the only way subscription status is going to change would be via Stripe webhooks. There is no way to change or cancel a subscription directly through our app. You have to go to the Stripe platform and make changes there, then those changes are reported to us via webhook.